flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointStatsTracker.java

```diff
@@ -227,11 +257,23 @@ void reportFailedCheckpoint(FailedCheckpointStats failed) {
             history.replacePendingCheckpointById(failed);
 
             dirty = true;
+            logCheckpointStatistics(failed);
         } finally {
             statsReadWriteLock.unlock();
         }
     }
 
+    private void logCheckpointStatistics(AbstractCheckpointStats checkpointStats) {
+        try {
+            StringWriter sw = new StringWriter();
+            MAPPER.writeValue(sw, CheckpointStatistics.generateCheckpointStatistics(checkpointStats, true));
+            String jsonDump = sw.toString();
+            LOG.debug("CheckpointStatistics (for jobID={}, jobName={}) dump = {} ", jobID, jobName, jsonDump);
+        } catch (Exception ex) {
+            LOG.error("Fail to log CheckpointStatistics", ex);
```

Comment:
nit: I'd log it as a warning

Reply:
I'd also add the job id and name to that message.